### DISCUSSION TITLE
About autoclosing PRs with different Helm values

### DISCUSSION THREAD

**Discussion by iamdmitrij** [renovatebot/renovate#24244] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.81.0

### If you're self-hosting Renovate, select which platform you are using.

Docker image on Azure DevOps (dev.azure.com)

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Perhaps it's possible to keep PRs open that don't match `helm-values.fileMatch` or `packageRules.matchFileNames` specific target files?

### What you tried so far.

I'm trying to use Renovate to update Helm value files per environment using same Docker Registry repository image dependency. I have Azure DevOps multi-stage pipeline which runs Renovate system to raise PRs to GitOps repo to update specific `value.<environment>.yaml` file per environment/stage. Docker image tag have semantic version.

The challenge here is that I have same registry and different Helm values files that need to be updated separately using different stages in the Azure DevOps pipeline.

Here is a part of my `renovate.config`

```
  "helm-values": {
    fileMatch: [`^${process.env.APP_NAME}\\/values\\.?(${process.env.APP_ENVIRONMENT})\\.ya?ml$`],
  },
  kubernetes: {
    fileMatch: ["(^|/)[^/]*\\.yaml$"],
  },
  additionalBranchPrefix: "{{packageFile}}-",

  packageRules: [
    {
      matchFileNames: [`${process.env.APP_NAME}/values.${process.env.APP_ENVIRONMENT}.yaml`],
      allowedVersions: process.env.APP_VERSION,
      labels: [process.env.APP_NAME, process.env.APP_ENVIRONMENT],
    }
  ],
```

My idea here was to only include certain `values.<environment>.yaml` file and upgrade it separately using specific `allowedVersions` semantic version that is generated from the pipeline stage. However, when running Renovate on already existing PRs (created by Renovate), Renovate autocloses PR from a different value file. For example, I've executed update on `values.staging.yaml` - there is a PR open, once I run a new pipeline on `values.development.yaml`, Renovate autocloses PR on staging file and opens PR with development only. Once a re-run previous stage (`values.staging.yaml` file), the roles are reversed and development PR is closed and staging is open.

Here are some debug logs from:

```
DEBUG: Branch lists (repository=GitOps)
       "branchList": [
         "renovate/poc-api/values.development.yaml-samplepoc.azurecr.io-ts-poc-api-0.1.x"
       ],
       "renovateBranches": [
         "renovate/poc-api/values.development.yaml-samplepoc.azurecr.io-ts-poc-api-0.1.x",
         "renovate/poc-api/values.staging.yaml-samplepoc.azurecr.io-ts-poc-api-0.1.x"
       ]
DEBUG: remainingBranches=renovate/poc-api/values.staging.yaml-samplepoc.azurecr.io-ts-poc-api-0.1.x (repository=GitOps)
DEBUG: getPrList() (repository=GitOps)
DEBUG: branch.isModified(): using git to calculate (repository=GitOps)
DEBUG: branch.isModified() = false (repository=GitOps)
DEBUG: setCachedModifiedResult(): Branch cache not present (repository=GitOps)
 INFO: Autoclosing PR (repository=GitOps)
       "branchName": "renovate/poc-api/values.staging.yaml-samplepoc.azurecr.io-ts-poc-api-0.1.x",
       "prNo": 8304,
       "prTitle": "Update samplepoc.azurecr.io/ts-poc-api Docker tag to v0.1.78"
DEBUG: updatePr(8304, Update samplepoc.azurecr.io/ts-poc-api Docker tag to v0.1.78 - autoclosed, body) (repository=GitOps)
DEBUG: Deleted remote branch: renovate/poc-api/values.staging.yaml-samplepoc.azurecr.io-ts-poc-api-0.1.x (repository=GitOps)

```

For some reason even if I specify that I want to update only that specific YAML file, it affect other YAML files in the repo PRs. Is there something that can be done here? The only other solution I can think of is using different Docker registry or repository to split environments.

**Comment by viceice** [comment]
You should not run multiple different renovate instances on one repo. you can use package rules to apply specific config options to specific paths.

**Reply by iamdmitrij** [reply]
If you mean running multiple instance at the same time, that's not what happens here. I run Renovate in stages, they won't be parallel. In my case, stages are executed in sequence: `development` -> `staging` - `production`.

For example, in `development` stage this configuration is executed:
```
  packageRules: [
    {
      matchFileNames: [`poc-api/values.development.yaml`],
      allowedVersions: "1.0.0",
      labels: ['poc-api', 'development'],
    }
  ],
```

and then in `staging` stage (after some time...):
```
  packageRules: [
    {
      matchFileNames: [`poc-api/values.staging.yaml`],
      allowedVersions: "1.0.0",
      labels: ['poc-api', 'staging'],
    }
  ],
```

What I don't completely understand why Renovate closes `development` PR when **only** second config is executed.

**Reply by viceice** [reply]
even in stages isn't supported
